### DISCUSSION TITLE
[@types/mapbox__mapbox-sdk] has incorrect result type for forwardGeocode

### DISCUSSION THREAD

**Discussion by public** [DefinitelyTyped/DefinitelyTyped#62116] [Issues with a @types package]
`forwardGeocode` is defined to return a ` MapiRequest<GeocodeResponse>`.

`GeocodeResponse` is an interface with a `features` attribute of type `GeocodeFeature[]`.

`GeocodeFeature` is an interface with a `context` attribute of type `GeocodeFeature[]`.

`GeocodeFeature` seems to not match the actual response type in the docs, or from real requests very well. For example this is what I really get in the response for a `feature`.

```
{
      id: 'address.6406825912128452',
      type: 'Feature',
      place_type: [ 'address' ],
      relevance: 1,
      properties: { accuracy: 'point' },
      text: 'Dunwellan Park',
      place_name: '38 Dunwellan Park, Shimna, Newcastle, BT33 0DD, United Kingdom',
      center: [ -5.892092, 54.216106 ],
      geometry: { type: 'Point', coordinates: [ -5.892092, 54.216106 ] },
      address: '38',
      context: [
        { id: 'postcode.6406825912128452', text: 'BT33 0DD' },
        { id: 'locality.158804559', text: 'Shimna' },
        { id: 'place.7866447', text: 'Newcastle' },
        { id: 'district.452175', wikidata: 'Q190684', text: 'Down' },
        {
          id: 'region.8061682358656870',
          short_code: 'GB-NIR',
          wikidata: 'Q26',
          text: 'Northern Ireland'
        },
        {
          id: 'country.12171874284814600',
          wikidata: 'Q145',
          short_code: 'gb',
          text: 'United Kingdom'
        }
      ]
    }
```

In particular `context` just doesn't seem to match at all really.

[a comment by an automated account is not shown]

**Comment by jeffbdye** [comment]
Hey @public, sorry for the delayed response - feel free to PR any changes to update the types to better match the documentation (and add a quick test)! I'd contributed these types awhile back with some help from the other contributors but used mostly the Directions service types and wasn't able to doublecheck the other services.
